### PR TITLE
Adding daily mail review from Issue 839

### PR DESCRIPTION
**Pull request by akmet**
I added the daily summery email as requested in #839.
I have not added any tests, because I never worked with this kind of tests and does not really know how to make them and what exact test for. Hope someone can help here?

Currently the script analyzes the /var/log/mail/mail.log and I think logrotate will cause a loss in information when it gets executed. One way to solve this problem could be running logrotate every day at midnight (what I am using in production system) and analyze the mail.log.1 which represents the day before.
Another way could be to analyze the combination of mail.log.1 and mail.log but this just doesn't feels right. Any thoughts?

### CODE REVIEW

File: target/bin/dailysummery [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+test -x /usr/sbin/pflogsumm || exit 0
+test -x /usr/bin/bsd-mailx || exit 0
+
+/usr/sbin/pflogsumm /var/log/mail/mail.log -d yesterday --problems-first | /usr/bin/bsd-mailx -s "Daily Summery for HOSTNAME"  DAILY_SUMM_EMAIL
```
Comment (johansmitsnl) [inline target/bin/dailysummery]: Should the variable be: `$DAILY_SUMM_EMAIL` ?

File: target/start-mailserver.sh [hunk 2]
```diff
@@ -1064,6 +1065,20 @@ function _setup_elk_forwarder() {
 		> /etc/filebeat/filebeat.yml
 }
 
+function _setup_daily_summery() {
+	notify 'task' 'Setting up daily summery'
+
+	DAILY_SUMM_EMAIL=${DAILY_SUMM_EMAIL:="0"}
```
Comment (johansmitsnl) [inline target/start-mailserver.sh]: The default should be set at the start of the script.

File: target/start-mailserver.sh [hunk 3]
```diff
@@ -1064,6 +1065,20 @@ function _setup_elk_forwarder() {
...
+	if [[ ! "$DAILY_SUMM_EMAIL" == 0 ]]; then
+		notify 'inf' "Enable daily mail summery with recipient $DAILY_SUMM_EMAIL"
+
+		sed -i -r 's/HOSTNAME/'$HOSTNAME'/g' /usr/local/bin/dailysummery
+		sed -i -r 's/DAILY_SUMM_EMAIL/'$DAILY_SUMM_EMAIL'/g' /usr/local/bin/dailysummery
+
+		echo "/usr/local/bin/dailysummery" > /etc/cron.daily/dailysummery
```
Comment (johansmitsnl) [inline target/start-mailserver.sh]: In the daily directory should contain executable scripts?
Just to confirm.

File: target/bin/dailysummery (hunk 1 above)
Comment (akmet) [reply target/bin/dailysummery]: I got problems with sed when used $. Removing it fixed the error..

**Comment by Starbix** [comment]
It'd be great if you could change `summery` to `summary`

**Comment by johansmitsnl** [comment]
@17Halbe could you assist with the tests?

File: target/bin/dailysummery (hunk 1 above)
Comment (17Halbe) [reply target/bin/dailysummery]: Even if you encapsulate the variable with `"`?

**Comment by 17Halbe** [comment]
Hey, what a great idea!
I'll be busy the next few days, but I'll give it a try.
What about sendmail instead of bsd-mailx wasn't it working properly?
I don't know pflogsumm, but if we already introduce a new env variable, what about letting the user decide between daily/weekly/monthly(?) mails?
The tests would need to confirm, that a mail is being sent by cron, right? That shouldn't be a problem.

File: Dockerfile [hunk 4]
```diff
@@ -21,6 +21,7 @@ RUN apt-get update -q --fix-missing && \
     amavisd-new \
     arj \
     binutils \
+    bsd-mailx \
```
Comment (17Halbe) [inline Dockerfile]: Isn't sendmail working for this?

Comment (akmet) [reply Dockerfile]: I wasn't able to do so. But maybe it was just my fault.
Could you provide an example on how to use sendmail? Is there something already existing here I could adapt?

Comment (17Halbe) [reply Dockerfile]: https://github.com/tomav/docker-mailserver/blob/a73692cc9fb98849bb0eac72311f4966ecaaca41/test/tests.bats#L1368
I haven't tried using it, but the tests seem to work if one is using it like this.. 
I don't like the way sendmail works either. It's different on every system, sometimes a symlink to some other mail service and then there are about a couple different implementations, all with different syntaxes... :roll_eyes:

**Comment by akmet** [comment]
@17Halbe Just had a look at manpage
It can just analyze today, yesterday or the complete log file.
But we could do this with logrotation and after each rotation we just analyze the old log and send an email.
What do you think?

**Comment by 17Halbe** [comment]
I think that would be overkill. ;) In that case, I think it‘s not worth the effort. 
Unless you like to do it for fun and practice! :) (Isn‘t that the reason we are all doing this anyway?)

**Comment by akmet** [comment]
When using in production we have to edit logrotate anyway because if not we are skipping one day as I mentioned above..

**Comment by akmet** [comment]
Travis failed cause because the postfix-accounts.cf file keeps removing while the container runs.
I can reproduce this on my desktop (win10 with docker running on Windows and Ubuntu Bash) and I don't understand why. Can someone reproduce this?

**Comment by johansmitsnl** [comment]
Is the file gone or is it empty?

**Comment by akmet** [comment]
Like both?
Windows Explorer shows the file, but I cannot open it ("File does not exist. Create File?") and I cannot write to it. The file only disappears when restarting the Docker Daemon.
I have no idea why this happens and I think that I then had to use my notebook for changes and git.

**Comment by johansmitsnl** [comment]
I don't know why that is, maybe change de filesystem of docker?

Could you also add a test by calling the function and verify the email has been send and it contains the correct subject and to address?

**Comment by 17Halbe** [comment]
I‘ll take care of the test. Probably today.

**Comment by akmet** [comment]
Me neither. But I installed Ubuntu next to Windows and started with a fresh master and now it seems to run.
I would also test if the interval is correctly set and if the logrotate config is set with defaults when no variable is set.

File: target/bin/dailysummery (hunk 1 above)
Comment (akmet) [reply target/bin/dailysummery]: Used the following line and it does not replace the variable:
sed -i -r 's/$HOSTNAME/'$HOSTNAME'/g' /usr/local/bin/postfix-summary
and
sed -i -r 's/\$HOSTNAME/'$HOSTNAME'/g' /usr/local/bin/postfix-summary

File: target/start-mailserver.sh (hunk 3 above)
Comment (akmet) [reply target/start-mailserver.sh]: due to using logrotate this is obsolet

**Comment by johansmitsnl** [comment]
Good work!

**Pull request merged**